src/java.base/share/classes/java/nio/charset/Charset.java

```diff
@@ -714,13 +713,8 @@ public final String name() {
      * @return  An immutable set of this charset's aliases
      */
     public final Set<String> aliases() {
-        if (aliasSet != null)
-            return aliasSet;
-        int n = aliases.length;
-        HashSet<String> hs = HashSet.newHashSet(n);
-        for (int i = 0; i < n; i++)
-            hs.add(aliases[i]);
-        aliasSet = Collections.unmodifiableSet(hs);
+        if (aliasSet == null)
+            aliasSet = Set.of(aliases);
```

Comment:
Given that this is racy then it might be better to use a local and avoid someone reading this code puzzling over whether it can return null, e.g.
```
        Set<String> aliasSet = this.aliasSet;
        if (aliasSet == null) {
            this.aliasSet = aliasSet = Set.of(aliases);
        }
```
You can also drop explicitly initialising it null at line 670.

Reply:
I have updated the code.

Reply:
> I have updated the code.

Thanks for the update. I think it looks okay although "as" looks a bit strange here, I think I would use aliasSet or something better.

Reply:
> > I have updated the code.
> 
> Thanks for the update. I think it looks okay although "as" looks a bit strange here, I think I would use aliasSet or something better.

Well, the name `as` is a bit strange. I didn't use the name `aliasSet` because I didn't want to shadow the field. 

Now I simply name it `set`. I think this name is clear enough in this context.